backend/inspirehep/migrator/tasks.py

```diff
@@ -504,11 +515,30 @@ def migrate_record_from_mirror(
         return record
 
 
-def wait_for_all_tasks(task):
+def wait_for_all_tasks(task, retry_count=0):
     if not task:
         return None
-    LOGGER.info(f"Waiting for task completion.", waiting_task_id=task)
-    next_task = AsyncResult(task).get()
+    LOGGER.info(
+        f"Waiting for task completion.", waiting_task_id=task, retry_count=retry_count
+    )
+    try:
+        next_task = AsyncResult(task).get()
+    except ResponseError:
+        if retry_count < MAX_RETRY_COUNT:
+            LOGGER.exception(
+                "Redis ResponseError Exception, retrying",
+                task=task,
+                retry_count=retry_count,
+            )
+            time.sleep(RETRY_BACKOFF * retry_count + 1)
+            return wait_for_all_tasks(task, retry_count + 1)
+        else:
+            LOGGER.exception(
+                "Redis ResponseError Exception failed too many times times.",
+                task=task,
```

Comment:
same

Reply:
Yes, this one we want to log as exception as it's failing here and stops after too many retries.

Reply:
I meant `task` -> `waiting_task_id` here

Reply:
:+1: